### PR TITLE
Refactor DigitalOcean modules

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
- Updated documentation fragment
- Updated Argument spec
- Use common REST class for API calls

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE

 - Feature Pull Request
 - Docs Pull Request

##### COMPONENT NAME

lib/ansible/module_utils/digital_ocean.py
lib/ansible/modules/cloud/digital_ocean/digital_ocean_block_storage.py
lib/ansible/modules/cloud/digital_ocean/digital_ocean_certificate.py
lib/ansible/modules/cloud/digital_ocean/digital_ocean_domain.py
lib/ansible/modules/cloud/digital_ocean/digital_ocean_floating_ip_facts.py
lib/ansible/modules/cloud/digital_ocean/digital_ocean_sshkey_facts.py
lib/ansible/modules/cloud/digital_ocean/digital_ocean_tag.py
lib/ansible/utils/module_docs_fragments/digital_ocean.py

##### ANSIBLE VERSION

```
2.5devel
```

### CODE REVIEW

File: lib/ansible/module_utils/digital_ocean.py [hunk 1]
```diff
@@ -96,6 +104,20 @@ def get_do_oauth_token(self):
             self.module.params.get('api_token') or \
             os.environ.get('DO_API_TOKEN') or \
             os.environ.get('DO_API_KEY') or \
-            os.environ.get('OAUTH_TOKEN')
+            os.environ.get('OAUTH_TOKEN') or \
+            os.environ.get('DO_OAUTH_TOKEN')
```
Comment (pmarques) [inline lib/ansible/module_utils/digital_ocean.py]: Is this necessary if we use are using **digital_ocean_argument_spec**?

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_floating_ip_facts.py [hunk 2]
```diff
@@ -98,41 +86,34 @@ def core(module):
     page = 1
     has_next = True
     floating_ips = []
-    while has_next or 200 != status_code:
+    status_code = 400
```
Comment (pmarques) [inline lib/ansible/modules/cloud/digital_ocean/digital_ocean_floating_ip_facts.py]: if you want to initialise wouldn't be better to set it to None?

**Review by pmarques — CHANGES_REQUESTED** [review]
@Akasurde just to comments, but LGTM

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_floating_ip_facts.py (hunk 2 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/digital_ocean/digital_ocean_floating_ip_facts.py]: ok.

File: lib/ansible/module_utils/digital_ocean.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/module_utils/digital_ocean.py]: ok.

**Review by pmarques — APPROVED** [review]
LGTM

**Comment by Akasurde** [comment]
@pmarques Thanks for review.

**Pull request merged**